client/rest-high-level/src/main/java/org/elasticsearch/client/rollup/RollableIndexCaps.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.client.rollup;
+
+import org.elasticsearch.common.ParseField;
+import org.elasticsearch.common.xcontent.ConstructingObjectParser;
+import org.elasticsearch.common.xcontent.ToXContentFragment;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Objects;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+/**
+ * Represents the rollup capabilities of a non-rollup index.  E.g. what values/aggregations
+ * were rolled up for this index, in what rollup jobs that data is stored and where those
+ * concrete rollup indices exist
+ *
+ * The index name can either be a single index, or an index pattern (logstash-*)
+ */
+public class RollableIndexCaps implements ToXContentFragment {
+    private static final ParseField ROLLUP_JOBS = new ParseField("rollup_jobs");
+
+    public static final Function<String, ConstructingObjectParser<RollableIndexCaps, Void>> PARSER = indexName -> {
+        @SuppressWarnings("unchecked")
+        ConstructingObjectParser<RollableIndexCaps, Void> p
+            = new ConstructingObjectParser<>(indexName,
+            a -> new RollableIndexCaps(indexName, (List<RollupJobCaps>) a[0]));
+
+        p.declareObjectArray(ConstructingObjectParser.constructorArg(), RollupJobCaps.PARSER::apply,
+            ROLLUP_JOBS);
+        return p;
+    };
+
+    private final String indexName;
+    private final List<RollupJobCaps> jobCaps;
+
+    RollableIndexCaps(final String indexName, final List<RollupJobCaps> caps) {
+        this.indexName = indexName;
+        this.jobCaps = Collections.unmodifiableList(Objects.requireNonNull(caps)
```

Comment:
It is... but isn't it better to sort once in the ctor instead of each time toXContent is called?  Since jobcaps is final and unmodifiable, we know it will only happen once and won't ever become unsorted.

Reply:
(also note iirc it's mainly there to make testing from yaml rest tests easier)

Reply:
Ok, let's keep the sorting in the ctor. I wanted to avoid the sort to be done each time the object is instantiated but this is a HLRC specific object so it will be done just one time